protoc-gen-gogo/generator/generator.go

```diff
@@ -1176,6 +1181,9 @@ func (g *Generator) addInitf(stmt string, a ...interface{}) {
 
 func (g *Generator) PrintImport(alias, pkg string) {
 	statement := "import " + alias + " " + strconv.Quote(pkg)
+	if alias == pkg[strings.LastIndex(pkg, "/")+1:] {
```

Comment:
I suspect this won't always work, since the packageName can differ from the packagePath.